tools/api/generate_go_protobuf.py

```diff
@@ -1,164 +1,13 @@
-        rule_dir, proto = rule.decode().rsplit('//', 1)[1].rsplit(':', 1)
-
-        prefix = '' if not api_repo else os.path.join('external', api_repo)
-        input_dir = os.path.join(bazel_bin, prefix, rule_dir, proto + '_', IMPORT_BASE, rule_dir)
-        input_files = glob.glob(os.path.join(input_dir, '*.go'))
-        output_dir = os.path.join(output, rule_dir)
-
-        # Ensure the output directory exists
-        os.makedirs(output_dir, 0o755, exist_ok=True)
-        for generated_file in input_files:
-            shutil.copy(generated_file, output_dir)
-    print('Go artifacts placed into: ' + output)
-
-
-def git(repo, *args):
-    cmd = ['git']
-    if repo:
-        cmd = cmd + ['-C', repo]
-    for arg in args:
-        cmd = cmd + [arg]
-    return check_output(cmd).decode()
-
-
-def clone_go_protobufs(repo):
-    # Create a local clone of go-control-plane
-    git(None, 'clone', 'git@github.com:envoyproxy/go-control-plane', repo, '-b', BRANCH)
-
-
-def find_last_sync_sha(repo):
-    # Determine last envoyproxy/envoy SHA in envoyproxy/go-control-plane
-    last_commit = git(repo, 'log', '--grep=' + MIRROR_MSG, '-n', '1', '--format=%B').strip()
-    # Initial SHA from which the APIs start syncing. Prior to that it was done manually.
-    if last_commit == "":
-        return 'e7f0b7176efdc65f96eb1697b829d1e6187f4502'
-    m = re.search(MIRROR_MSG + '(\w+)', last_commit)
-    return m.group(1)
-
-
-def updated_since_sha(repo, last_sha):
-    # Determine if there are changes to API since last SHA
-    return git(None, 'rev-list', '%s..HEAD' % last_sha).split()
-
-
-def write_revision_info(repo, sha):
-    # Put a file in the generated code root containing the latest mirrored SHA
-    dst = os.path.join(repo, 'envoy', 'COMMIT')
-    with open(dst, 'w') as fh:
-        fh.write(sha)
-
-
-def sync_go_protobufs(output, repo):
-    for folder in ['envoy', 'contrib']:
-        # Sync generated content against repo and return true if there is a commit necessary
-        dst = os.path.join(repo, folder)
-        # Remove subtree in repo
-        git(repo, 'rm', '-r', '--ignore-unmatch', folder)
-        # Copy subtree from output to repo
-        shutil.copytree(os.path.join(output, folder), dst)
-        git(repo, 'add', folder)
-
-
-def publish_go_protobufs(repo, sha):
-    # Publish generated files with the last SHA changes to API
-    git(repo, 'config', 'user.name', USER_NAME)
-    git(repo, 'config', 'user.email', USER_EMAIL)
-    git(repo, 'add', 'envoy')
-    git(repo, 'add', 'contrib')
-    git(repo, 'commit', '--allow-empty', '-s', '-m', MIRROR_MSG + sha)
-    git(repo, 'push', 'origin', BRANCH)
-
-
-def updated(repo):
-    return len([
-        f for f in git(repo, 'diff', 'HEAD', '--name-only').splitlines() if f != 'envoy/COMMIT'
-    ]) > 0
+def main():
+    # TODO(phlax): remove this 12/23
+    sys.stderr.write("This script has been removed, please use `ci/do_ci.sh api.go` instead\n")
```

Comment:
Should we just delete this script? Less cruft n the tools folder

Reply:
im a little concerned that others may be using it so would rather add this ~deprecation/removal notice for now